client/pages/queue/[section].vue

```diff
@@ -344,31 +361,43 @@ const filteredDocuments = computed(() => {
     case 'enqueuing':
       docs = documents.value.filter((d: any) => d.disposition === 'created')
       break
-    case 'pending':
-      docs = documents.value.filter(
-        (d: any) =>
-          d.disposition === 'in_progress' && d.assignmentSet?.length === 0
-      )
-      break
-    case 'exceptions':
-      docs = documents.value.filter(
-        (d: any) =>
-          d.disposition === 'in_progress' &&
-          d.labels?.filter((lbl: any) => lbl.isException).length
-      )
-      break
-    case 'inprocess':
+    case 'queue':
       docs = documents.value
         .filter(
           (d: any) =>
-            d.disposition === 'in_progress' && d.assignmentSet?.length > 0
+            d.disposition === 'in_progress'
         )
+        .filter(
+          (d: any) => {
+            const needsAssignmentFilterFn = () => {
+              if(needsAssignmentTristate.value === true) {
+                return Boolean(!d.assignmentSet || d.assignmentSet.length === 0)
+              } else if(needsAssignmentTristate.value === false) {
+                return Boolean(d.assignmentSet?.length > 0)
+              } else if(needsAssignmentTristate.value === CHECKBOX_INDETERMINATE) {
+                return true
+              }
+            }
+
+            const hasExceptionFilterFn = () => {
+              if(hasExceptionTristate.value === true) {
+                return Boolean(d.exception)
```

Comment:
Is `exception` getting set somewhere? It should be a check for an attached label with `is_exception` set. The back end doesn't currently do this. Should it?

I lean toward yes, probably as a `hasException` field, but that raises the question of whether it should also have an `exceptions` list. I'd rather not because it would (I guess) be redundant with the `labels` list... any thoughts?

Reply:
@jennifer-richards I've updated the filter again. Is that what it should look like? We can do the server approach instead if you want.

Reply:
lgtm, and it's simple enough I see no need to complicate it futher :-)